rust/connlib/shared/src/proptest.rs

```diff
@@ -1,52 +1,104 @@
 use crate::messages::{
     client::ResourceDescriptionCidr,
-    client::{GatewayGroup, ResourceDescriptionDns, SiteId},
-    ClientId, ResourceId,
+    client::{GatewayGroup, ResourceDescription, ResourceDescriptionDns, SiteId},
+    ClientId, GatewayId, ResourceId,
 };
 use ip_network::{IpNetwork, Ipv4Network, Ipv6Network};
+use itertools::Itertools;
 use proptest::{
     arbitrary::{any, any_with},
     collection, sample,
-    strategy::Strategy,
+    strategy::{Just, Strategy},
 };
 use std::net::{Ipv4Addr, Ipv6Addr};
 
-pub fn dns_resource() -> impl Strategy<Value = ResourceDescriptionDns> {
+// Generate resources sharing 1 gateway group
```

Comment:
Can we call this "site" please? It would be good to stick to one language.

Reply:
Seconded, should just do the mapping when desrializing from portal and use `site` every in Rust land